hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java

```diff
@@ -431,9 +430,12 @@ public void stop() {
 
 	protected void stopCacheRegions() {
 		log.debug( "Clear region references" );
-		getCacheCommandFactory( manager.getCache().getAdvancedCache() )
-				.clearRegions( regionNames );
-		regionNames.clear();
+		getCacheCommandFactory().clearRegions( regions );
+		// Ensure we cleanup any caches we created
+		for ( BaseRegion region : regions ) {
+			region.getCache().stop();
```

Comment:
@rvansa EmbeddedCacheManager.undefineConfiguration(...) wasn't added until Infinispan 8.0.x.  The Hibernate 5.0.x branch requires Infinispan 7.2.x.